### DISCUSSION TITLE
Roles for Users

### DISCUSSION THREAD

**Discussion by mountiny** [supabase/supabase#458] [Questions] [answered]
Hi! I am working on one simple app using your great Stripe, Next.js Starter. I am now stuck with one problem. 

For my app, I need two roles for the users I create, let's say a business and buyer roles. I have different pages on frontend to create buyer user account and to create a business user account. 

However, I struggle to find a way how to create these in database as the call will create generic user and the confirmation email takes the user back to my website with no way for me to distinguish, whether it was business or buyer user. Even if I would use localStorage, I can't assume the user will click the confirmation link in email on the same device or use the same browser for it.

Do  you have any idea or tips how to achieve this workflow? 

Thank you very much!

**Comment by kiwicopple** [comment, marked as answer]
Are you saving the user to the public schema too ? https://supabase.io/docs/guides/auth#create-a-publicusers-table

If you are then you can probably do this in your signup function. Something like this

```js
const signUpUser = async (email, password, role) => {

  const { user, error } = await supabase
    .auth
    .signUp({ email, password})
    
  const { data, error } = await supabase
    .from('user')
    .update({ role })
    .eq('id', user.id) 
}
```

Hope that helps!

**Reply by mountiny** [reply]
@kiwicopple Hi! Thank you very much! I am actually using your Stripe / Next.js starter, so there is users public table. 

So if I understand it right, even if the user email confirmation is enabled, the signUp function will return user straight after creating the instance in the users table? I thought I wont be able to get the user instance before user would confirm his email.

**Reply by kiwicopple** [reply]
Yes it will still create and return the user! If you run `select * from auth.users;` you will see that there is a `confirmed_at` field for tracking confirmed users. It will be empty for unconfirmed users

**Reply by mountiny** [reply]
Ah, thank you very much! I don't know why I thought the user is not returned until they are confirmed... Thanks for help!

**Reply by mountiny** [reply]
@kiwicopple Hi! I am sorry for actually takin gos long, but I have not got a chance to get back to this for a while. If I use the approach you have suggested and I have the email confirmation on which I want to keep it that way, then the user returned my signUp function is null and I get the user instance only once I follow the Sign up confirmation link. 

Do you think that the Stripe Next.js starter might be set up this way with the public users table?

Is there a way I could get the user instance before they confirm the email? 

Thank you for the help!

**Reply by roker15** [reply]
> Yes it will still create and return the user! If you run `select * from auth.users;` you will see that there is a `confirmed_at` field for tracking confirmed users. It will be empty for unconfirmed users

I tried same with magic link user not returned if i do not confirmed email 

```
`const { user, error } = await supabase.auth.signIn({ email });
      console.log(console.log(user))
```
